test/test_functionalization.py

```diff
@@ -27,25 +27,29 @@ def are_aliased(x, y):
 # if/when functorch moves into core.
 # This is basically a crappy version of `functionalize()` for single-tensor-arg inputs.
 def _functionalize(f, *, reapply_views: bool, crossref: bool):
-    def wrapped(a):
+    def wrapped(*inputs):
         ctx = nullcontext()
         if crossref:
             ctx = enable_crossref_functionalize()
         with ctx:
-            input_functional = torch._to_functional_tensor(a)
-            input_functional.requires_grad = a.requires_grad
+            inputs_functional = []
+            for inpt in inputs:
+                input_functional = torch._to_functional_tensor(inpt)
+                input_functional.requires_grad = inpt.requires_grad
+                inputs_functional.append(input_functional)
```

Comment:
nit- you can support arbitrarily nested containers with pytrees:

`inputs_functional = pytree.tree_map_only(torch.Tensor, to_fun)` (where `to_fun` is a helper that does the wrapping and metadata setting)